src/engine/ebur128analysis.cpp

```diff
@@ -0,0 +1,503 @@
+// *     |Elevation (phi)| < 30deg
+// * and 60deg <= |Azimuth (theta)| <= 120° (i.e. +-90deg +- 30deg)
+// ... then the channel is weighted at +1.5 dB.
+//
+// ITU R-REC-BS 1770-4 uppper and bottom position channels are at +-45deg,
+// So only the middle-position channels are affected.
+channel gst_channel_to_ebur_channel(GstAudioChannelPosition pos) {
+
+  switch (pos) {
+    case GST_AUDIO_CHANNEL_POSITION_NONE:
+    case GST_AUDIO_CHANNEL_POSITION_INVALID:
+    case GST_AUDIO_CHANNEL_POSITION_LFE1:
+    case GST_AUDIO_CHANNEL_POSITION_LFE2:
+      return EBUR128_UNUSED;
+
+    case GST_AUDIO_CHANNEL_POSITION_MONO:
+      return EBUR128_DUAL_MONO; // +6 dB
+
+    case GST_AUDIO_CHANNEL_POSITION_SURROUND_LEFT:
+      return EBUR128_LEFT_SURROUND;
+    case GST_AUDIO_CHANNEL_POSITION_SURROUND_RIGHT:
+      return EBUR128_RIGHT_SURROUND;
+
+    case GST_AUDIO_CHANNEL_POSITION_BOTTOM_FRONT_CENTER:
+      return EBUR128_Bp000;
+
+    case GST_AUDIO_CHANNEL_POSITION_BOTTOM_FRONT_LEFT:
+      return EBUR128_Bm045;
+    case GST_AUDIO_CHANNEL_POSITION_BOTTOM_FRONT_RIGHT:
+      return EBUR128_Bp045;
+
+    case GST_AUDIO_CHANNEL_POSITION_TOP_CENTER:
+    case GST_AUDIO_CHANNEL_POSITION_TOP_FRONT_CENTER:
+      return EBUR128_Up000;
+
+    case GST_AUDIO_CHANNEL_POSITION_TOP_REAR_CENTER:
+      return EBUR128_Up180;
+
+    case GST_AUDIO_CHANNEL_POSITION_TOP_SIDE_LEFT:
+      return EBUR128_Um090;
+    case GST_AUDIO_CHANNEL_POSITION_TOP_SIDE_RIGHT:
+      return EBUR128_Up090;
+
+    case GST_AUDIO_CHANNEL_POSITION_TOP_FRONT_LEFT:
+      return EBUR128_Um045;
+    case GST_AUDIO_CHANNEL_POSITION_TOP_FRONT_RIGHT:
+      return EBUR128_Up045;
+
+    case GST_AUDIO_CHANNEL_POSITION_TOP_REAR_LEFT:
+      return EBUR128_Um135;
+    case GST_AUDIO_CHANNEL_POSITION_TOP_REAR_RIGHT:
+      return EBUR128_Up135;
+
+    case GST_AUDIO_CHANNEL_POSITION_FRONT_CENTER:
+      return EBUR128_Mp000;
+
+    case GST_AUDIO_CHANNEL_POSITION_REAR_CENTER:
+      return EBUR128_Mp180;
+
+    case GST_AUDIO_CHANNEL_POSITION_FRONT_LEFT:
+      return EBUR128_Mm030;
+    case GST_AUDIO_CHANNEL_POSITION_FRONT_RIGHT:
+      return EBUR128_Mp030;
+
+    case GST_AUDIO_CHANNEL_POSITION_FRONT_LEFT_OF_CENTER:
+      return EBUR128_MmSC;
+    case GST_AUDIO_CHANNEL_POSITION_FRONT_RIGHT_OF_CENTER:
+      return EBUR128_MpSC;
+
+    case GST_AUDIO_CHANNEL_POSITION_WIDE_LEFT:
+      return EBUR128_Mm060; // +1.5 dB
+    case GST_AUDIO_CHANNEL_POSITION_WIDE_RIGHT:
+      return EBUR128_Mp060; // +1.5 dB
+
+    case GST_AUDIO_CHANNEL_POSITION_SIDE_LEFT:
+      return EBUR128_Mm090; // +1.5 dB
+    case GST_AUDIO_CHANNEL_POSITION_SIDE_RIGHT:
+      return EBUR128_Mp090; // +1.5 dB
+
+    case GST_AUDIO_CHANNEL_POSITION_REAR_LEFT:
+      return EBUR128_Mm110; // +1.5 dB
+    case GST_AUDIO_CHANNEL_POSITION_REAR_RIGHT:
+      return EBUR128_Mp110; // +1.5 dB
+  }
+
+}
+
+struct FrameFormat {
+  enum class DataFormat {
+    S16,
+    S32,
+    FP32,
+    FP64
+  };
+
+  int channels;
```

Comment:
Initialize these in an initialization list

Reply:
I think i'm missing a point. Why should these be initialized?
`FrameFormat` can only be constructed via `explicit FrameFormat(GstCaps *caps);`,
it can't be default-constructed in any way, so we can't ever end up
with an uninitialized fields anyway.